lib/ConstExtract/ConstExtract.cpp

```diff
@@ -489,14 +500,45 @@ void writeAttributes(
   });
 }
 
+void writeResultBuilderInformation(llvm::json::OStream &JSON,
+                                   const swift::NominalTypeDecl *TypeDecl,
+                                   const swift::VarDecl *VarDecl) {
+  if (auto *attr = VarDecl->getAttachedResultBuilder()) {
+    JSON.attributeObject("resultBuilder", [&] {
+      JSON.attribute("type", toFullyQualifiedTypeNameString(attr->getType()));
+      writeFileInformation(JSON, VarDecl->getASTContext(), attr);
+    });
+
+    return;
+  }
+
+  for (ProtocolDecl *Decl :
+       TypeDecl->getLocalProtocols(ConformanceLookupKind::All)) {
+    for (auto Member : Decl->getMembers()) {
+      if (auto *VD = dyn_cast<swift::VarDecl>(Member)) {
+        if (auto *attr = VD->getAttachedResultBuilder()) {
+          if (VD->getName() == VarDecl->getName()) {
```

Comment:
Nit: `getAttachedResultBuilder` fires off a request, so I think checking if the `VD->getName()` is the one we're looking for *first* could save some unnecessary work for all the other properties.

Reply:
Makes sense to me. I'll change that order and add the early return.